diff --git a/.ci/manywheel/build.sh b/.ci/manywheel/build.sh
index 61beb47..25f3b33 100755
--- a/.ci/manywheel/build.sh
+++ b/.ci/manywheel/build.sh
@@ -11,7 +11,7 @@ case "${GPU_ARCH_TYPE:-BLANK}" in
     rocm)
         bash "${SCRIPTPATH}/build_rocm.sh"
         ;;
-    cpu | cpu-cxx11-abi | cpu-aarch64 | cpu-s390x)
+    cpu | cpu-cxx11-abi | cpu-aarch64 | cpu-s390x | cpu-loongarch64)
         bash "${SCRIPTPATH}/build_cpu.sh"
         ;;
     xpu)
diff --git a/.ci/manywheel/build_common.sh b/.ci/manywheel/build_common.sh
index d50bd62..b288e65 100644
--- a/.ci/manywheel/build_common.sh
+++ b/.ci/manywheel/build_common.sh
@@ -35,12 +35,17 @@ if [[ "$OS_NAME" == *"AlmaLinux"* ]]; then
         aarch64)
             PLATFORM="manylinux_2_28_aarch64"
             ;;
+        loongarch64)
+            PLATFORM="manylinux_2_38_loongarch64"
+            ;;
         *)
             echo "Other architectures: $ARCH, not setting PLATFORM"
             ;;
     esac
 elif [[ "$OS_NAME" == *"Red Hat Enterprise Linux"* ]]; then
     retry dnf install -q -y zip openssl
+elif [[ "$OS_NAME" == *"Anolis OS"* ]]; then
+    retry dnf install -q -y zip openssl
 elif [[ "$OS_NAME" == *"Ubuntu"* ]]; then
     # TODO: Remove this once nvidia package repos are back online
     # Comment out nvidia repositories to prevent them from getting apt-get updated, see https://github.com/pytorch/pytorch/issues/74968
@@ -114,6 +119,7 @@ if [[ -z "$PYTORCH_ROOT" ]]; then
     exit 1
 fi
 pushd "$PYTORCH_ROOT"
+retry pip install -U pip
 retry pip install -qUr requirements-build.txt
 python setup.py clean
 retry pip install -qr requirements.txt
@@ -371,6 +377,10 @@ for pkg in /$WHEELHOUSE_DIR/torch_no_python*.whl /$WHEELHOUSE_DIR/torch*linux*.w
         wheel_file=$(echo $(basename $pkg) | sed -e 's/-cp.*$/.dist-info\/WHEEL/g')
         sed -i -e s#linux_aarch64#"${PLATFORM}"# $wheel_file;
     fi
+    if [[ $PLATFORM == "manylinux_2_38_loongarch64" ]]; then
+        wheel_file=$(echo $(basename $pkg) | sed -e 's/-cp.*$/.dist-info\/WHEEL/g')
+        sed -i -e s#linux_loongarch64#"${PLATFORM}"# $wheel_file;
+    fi
 
     # regenerate the RECORD file with new hashes
     record_file=$(echo $(basename $pkg) | sed -e 's/-cp.*$/.dist-info\/RECORD/g')
@@ -422,6 +432,11 @@ for pkg in /$WHEELHOUSE_DIR/torch_no_python*.whl /$WHEELHOUSE_DIR/torch*linux*.w
         zip -rq $pkg_name $PREIX*
         rm -f $pkg
         mv $pkg_name $(dirname $pkg)/$pkg_name
+    elif [[ $PLATFORM == "manylinux_2_38_loongarch64" ]]; then
+        pkg_name=$(echo $(basename $pkg) | sed -e s#linux_loongarch64#"${PLATFORM}"#)
+        zip -rq $pkg_name $PREIX*
+        rm -f $pkg
+        mv $pkg_name $(dirname $pkg)/$pkg_name
     else
         # zip up the wheel back
         zip -rq $(basename $pkg) $PREIX*
diff --git a/.ci/manywheel/build_cpu.sh b/.ci/manywheel/build_cpu.sh
index 9a6b14c..6adfd1f 100755
--- a/.ci/manywheel/build_cpu.sh
+++ b/.ci/manywheel/build_cpu.sh
@@ -62,6 +62,8 @@ elif [[ "$OS_NAME" == *"Red Hat Enterprise Linux"* ]]; then
     LIBGOMP_PATH="/usr/lib64/libgomp.so.1"
 elif [[ "$OS_NAME" == *"AlmaLinux"* ]]; then
     LIBGOMP_PATH="/usr/lib64/libgomp.so.1"
+elif [[ "$OS_NAME" == *"Anolis OS"* ]]; then
+    LIBGOMP_PATH="/usr/lib64/libgomp.so.1"
 elif [[ "$OS_NAME" == *"Ubuntu"* ]]; then
     if [[ "$ARCH" == "s390x" ]]; then
         LIBGOMP_PATH="/usr/lib/s390x-linux-gnu/libgomp.so.1"
diff --git a/.ci/pytorch/build.sh b/.ci/pytorch/build.sh
index eb3529a..70be143 100755
--- a/.ci/pytorch/build.sh
+++ b/.ci/pytorch/build.sh
@@ -248,7 +248,7 @@ fi
 
 # Do not change workspace permissions for ROCm and s390x CI jobs
 # as it can leave workspace with bad permissions for cancelled jobs
-if [[ "$BUILD_ENVIRONMENT" != *rocm* && "$BUILD_ENVIRONMENT" != *s390x* && "$BUILD_ENVIRONMENT" != *riscv64* && -d /var/lib/jenkins/workspace ]]; then
+if [[ "$BUILD_ENVIRONMENT" != *rocm* && "$BUILD_ENVIRONMENT" != *loongarch64* && "$BUILD_ENVIRONMENT" != *s390x* && "$BUILD_ENVIRONMENT" != *riscv64* && -d /var/lib/jenkins/workspace ]]; then
   # Workaround for dind-rootless userid mapping (https://github.com/pytorch/ci-infra/issues/96)
   WORKSPACE_ORIGINAL_OWNER_ID=$(stat -c '%u' "/var/lib/jenkins/workspace")
   cleanup_workspace() {
@@ -445,6 +445,6 @@ if [[ "$BUILD_ENVIRONMENT" != *libtorch* && "$BUILD_ENVIRONMENT" != *bazel* ]];
   PYTHONPATH=. python tools/stats/export_test_times.py
 fi
 # don't do this for bazel or s390x or riscv64 as they don't use sccache
-if [[ "$BUILD_ENVIRONMENT" != *s390x* && "$BUILD_ENVIRONMENT" != *riscv64* && "$BUILD_ENVIRONMENT" != *-bazel-* ]]; then
+if [[ "$BUILD_ENVIRONMENT" != *loongarch64* "$BUILD_ENVIRONMENT" != *s390x* && "$BUILD_ENVIRONMENT" != *riscv64* && "$BUILD_ENVIRONMENT" != *-bazel-* ]]; then
   print_sccache_stats
 fi
diff --git a/requirements.txt b/requirements.txt
index ae7f335..b4313f3 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -10,7 +10,7 @@ filelock
 fsspec>=0.8.5
 hypothesis
 jinja2
-lintrunner ; platform_machine != "s390x"
+lintrunner ; platform_machine != "loongarch64"
 networkx>=2.5.1
 optree>=0.13.0
 psutil
